### PR TITLE
Fix linting

### PR DESCRIPTION
**Pull request by lalexgap**
It looks like our previous eslint file had references to various metamask plugins (like `@metamask/eslint-config-typescript`) that we never imported from in our `package.json`. That meant trying to run the linter would crash, as the metamask depenencies were missing.

This PR addresses this by:
- Removing references to metamask in our eslint file. Instead I've just copied the eslint config file from nitro protocol.
- Now that the linter is working I've run it and cleaned up a couple of files
- I've added a root level `build` and `lint` command
- I've added a CI workflow to `build` and `lint` everything.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,44 +1,67 @@
...
+          "sibling",
+          "index",
+        ],
+        "newlines-between": "always",
+      },
+    ],
+    /**
+     * The default setting for Prettier is 'warn' because then it shows as yellow squiggly lines
+     * in the VS Code IDE. However, it means `eslint` will not have an error code if there is warning
+     * due to prettier unles you also add the `--max-warnings=0` flag in front of it. So, in the `lint-staged`
```
Comment (kerzhner) [inline .eslintrc.js]:
```suggestion
     * due to prettier unless you also add the `--max-warnings=0` flag in front of it. So, in the `lint-staged`
```

File: .eslintrc.js [hunk 2]
```diff
@@ -1,44 +1,67 @@
...
+     * scripts in the packages within this monorepo, we add that flag so that the precommit hooks
```
Comment (kerzhner) [inline .eslintrc.js]: Do we understand this comment? Does this comment hold true for the `nitro-gui` repo?

File: .eslintrc.js [hunk 3]
```diff
@@ -1,44 +1,67 @@
...
-    },
-
-    {
-      files: ['**/*.{ts,tsx}'],
-      extends: ['@metamask/eslint-config-typescript'],
+      files: ["**/*.ts"],
+      extends: ["plugin:@typescript-eslint/recommended"],
       rules: {
-        '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+        "@typescript-eslint/no-explicit-any": "off",
```
Comment (kerzhner) [inline .eslintrc.js]: Any particular reason to turn off `no-explicit-any`?

File: .eslintrc.js [hunk 4]
```diff
@@ -1,44 +1,67 @@
...
     {
-      files: ['**/*.test.ts', '**/*.test.js'],
-      extends: ['@metamask/eslint-config-jest'],
+      files: ["./src/**/*.ts"],
       rules: {
-        '@typescript-eslint/no-shadow': [
-          'error',
-          { allow: ['describe', 'expect', 'it'] },
-        ],
+        "@typescript-eslint/no-explicit-any": "warn",
```
Comment (kerzhner) [inline .eslintrc.js]: Should we try `error` for this setting and see if we can work with the stricter lint rules?

File: packages/nitro-rpc-client/.eslintrc.js
Comment (kerzhner) [inline packages/nitro-rpc-client/.eslintrc.js:1]: Does `eslintrc.js` need to be kept around in the `site` package?

**Review by kerzhner — APPROVED** [review]
Thanks for tackling linting!

File: packages/nitro-rpc-client/.eslintrc.js
Comment (lalexgap) [reply packages/nitro-rpc-client/.eslintrc.js:1]: > Does `eslintrc.js` need to be kept around in the `site` package?

It looks like it extends some react specific rules, so  think so.

File: .eslintrc.js (hunk 4 above)
Comment (lalexgap) [reply .eslintrc.js]: https://github.com/statechannels/nitro-gui/pull/19/commits/db5898edf20411442fd3c6d65cd6fd580d346218

File: .eslintrc.js (hunk 3 above)
Comment (lalexgap) [reply .eslintrc.js]: Removed in https://github.com/statechannels/nitro-gui/pull/19/commits/db5898edf20411442fd3c6d65cd6fd580d346218

File: .eslintrc.js (hunk 2 above)
Comment (lalexgap) [reply .eslintrc.js]: Removed in https://github.com/statechannels/nitro-gui/pull/19/commits/db5898edf20411442fd3c6d65cd6fd580d346218

**Pull request merged**